odxtools/diaglayer.py

```diff
@@ -273,74 +199,28 @@ def finalize_init(self, odxlinks: Optional[OdxLinkDatabase] = None):
 
     def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
         """Construct a mapping from IDs to all objects that are contained in this diagnostic layer."""
-        logger.info(f"Adding {self.odx_id} to odxlinks.")
-
-        odxlinks = {self.odx_id: self}
-
-        for obj in chain(
-                self._local_services,
-                self._local_single_ecu_jobs,
-                self.requests,
-                self.positive_responses,
-                self.negative_responses,
-                self.additional_audiences,
-                self.functional_classes,
-                self.state_charts,
-        ):
-            odxlinks[obj.odx_id] = obj
-
-        for obj in chain(
-                self._local_services,
-                self._local_single_ecu_jobs,
-                self.requests,
-                self.positive_responses,
-                self.negative_responses,
-                self.additional_audiences,
-                self.functional_classes,
-                self.sdgs,
-                self.state_charts,
-        ):
-            odxlinks.update(obj._build_odxlinks())
+        result = self.diag_layer_raw._build_odxlinks()
 
-        if self.local_diag_data_dictionary_spec:
-            odxlinks.update(self.local_diag_data_dictionary_spec._build_odxlinks())
+        # we want to get the full diag layer, not just the raw layer
+        # when referencing...
+        result[self.odx_id] = self
 
-        return odxlinks
+        return result
 
     def _resolve_references(self, odxlinks: OdxLinkDatabase) -> None:
```

Comment:
Looking at _resolve_references and _build_odxlinks, I am more convinced that you should just use class inheritance